bothan-coingecko/src/api/rest.rs

```diff
@@ -21,57 +20,56 @@ impl CoinGeckoRestAPI {
     pub async fn get_coins_list(&self) -> Result<Vec<Coin>, RestAPIError> {
         let url = format!("{}coins/list", self.url);
         let builder = self.client.get(url);
-        let response = send_request(builder).await?;
 
-        Ok(response.json::<Vec<Coin>>().await?)
+        request::<Vec<Coin>>(builder).await
     }
 
     /// Retrieves market data for the specified coins from the CoinGecko API.
-    pub async fn get_coins_market(
+    pub async fn get_coins_market<T: AsRef<str>>(
         &self,
-        ids: &[&str],
-        page_size: usize,
-        page: usize,
-    ) -> Result<Vec<Option<Market>>, RestAPIError> {
+        ids: &[T],
+        order: Option<Order>,
+        page_size: Option<usize>,
+        page: Option<usize>,
+    ) -> Result<Vec<Market>, RestAPIError> {
         let url = format!("{}coins/markets", self.url);
```

Comment:
What 's the different between these 2 endpoints?
1. https://docs.coingecko.com/reference/simple-price
2. https://docs.coingecko.com/reference/coins-markets

Reply:
coins-market is more verbose, but since simple-price is newer and doesn't require pagination, will switch to that instead